lib/msal-browser/docs/accounts.md

```diff
@@ -35,25 +57,79 @@ function handleResponse(resp) {
 }
 ```
 
-Now the `homeAccountId`, `localAccountId`, or `username` can be used to look up the cached account before acquiring a token silently:
+Now, account properties such as: `homeAccountId`, `localAccountId`, and `username` can be used to look up the cached account before acquiring a token silently:
 
 ```javascript
 // This method attempts silent token acquisition and falls back on acquireTokenPopup
-async function getTokenPopup(request, account) {
-    request.account = myMSALObj.getAccountByHomeId(homeAccountId); // alternatively: myMSALObj.getAccountByLocalId(localAccountId) or myMSALObj.getAccountByUsername(username) 
+async function getTokenPopup(request, homeAccountId) {
+    // In this case, accounts are filtered by homeAccountId, but more attributes can be added to refine the search and increase the precision of the account filter
+    const accountFilter = {
+        homeAccountId: homeAccountId,
+    };
+    request.account = myMSALObj.getAccountByFilter(accountFilter);
     return await myMSALObj.acquireTokenSilent(request).catch(async (error) => {
-       // Handle error
+        // Handle error
         return await myMSALObj.acquireTokenPopup(request);
     });
 }
 ```
 
+### Filtering by login hint
+
+As of `@azure/msal-browser@3.11.0`, all login hint values can be used to search for and filter accounts. In order to filter by login hint, MSAL will compare the `loginHint` value in the `AccountFilter` object against the following account attributes (in order of precedence) to search for matches:
+
+-   `login_hint` ID token claim
+-   `username` account property
+-   `upn` ID token claim
+
+> Note: Regardless of which attribute above is picked, they must be passed into the account filter as `loginHint`.
+
+#### Using `login_hint` claim
+
+```javascript
+const accountFilter = {
+    loginHint: previouslyObtainedIdTokenClaims.login_hint;
+};
+request.account = myMSALObj.getAccountByFilter(accountFilter);
+return await myMSALObj.acquireTokenSilent(request).catch(async (error) => {
+    // Handle error
+    return await myMSALObj.acquireTokenPopup(request);
+});
+```
+
+#### Using `username``
+
+```javascript
+const accountUsername = userProfile.username;
+const accountFilter = {
+    loginHint: accountUsername;
```

Comment:
Based on the AccountFilter type above, could you also provide this as `username`?

Reply:
Yes you could, that was already enabled though. Considering it in login hint is mostly useful if you pick a login_hint out of whichever is available between the three and then stop caring what the source of the login_hint was (in case you lose context). So 10k lines above const loginHint was assigned to something, but you know for sure you can pass it into getAccount and it'll get compared to all the options available.

Reply:
That's fair, if you know you have a username however, providing it as username will be slightly more performant. We should call out both so folks don't think they must provide it as loginHint

Reply:
I added some context/disambiguation.